architecture/2024-08-dashboard-component-threat-model.md

```diff
@@ -0,0 +1,299 @@
+   
+**Status**:
+
+Active. Operators are expected to configure the [Backstage  permissions system](https://backstage.io/docs/permissions/overview) to restrict access as necessary. 
+
+2. Audit logs should be enabled to monitor and report on suspicious user activity. 
+
+**Status**
+
+[In Progress](https://github.com/backstage/backstage/issues/23950)
+
+#### Threat 3: Spoofing dashboard service-account can cause DoS 
+**Description**
+
+If an unauthorized user or malicious actor gtampers with cluster,
+he can create services with dashboard service-account to send Radius API requests. 
+
+**Impact**:
+
+A malicious user can utilize Radius API and create too many resources. 
+
+This is because dashboard service-account is allowed get, list, and post requests to UCP
+
+```
+metadata:
+  name: dashboard
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: dashboard
+    app.kubernetes.io/part-of: radius
+rules:
+  - apiGroups: ['api.ucp.dev']
+    resources: ['*']
+    # dashboard needs get, list, and post privileges for api.ucp.dev
+    verbs: ['*']
+```
+
+**Mitigation**:
+
+Tampering with the dashboard service-account or creating malicious services would require access to modify the radius-system namespace. Our threat model assumes that the operator has limited access to the radius-system namespace using Kubernetes' existing RBAC mechanism.
+   
+**Status**:
+
+Active. Operators are expected to secure their cluster and limit access to the radius-system namespace. 
+
+#### Threat 3: Sniffing traffic from dashboard server to browser could result in Information Disclosure. 
+
+If HTTPS is not enabled and the dashboard is made public-facing then a malicious user would be able to see the data used to construct dashboard by sniffing.
+
+**Impact**:
+
+If an unauthorized user or malicious actor gathers data used to construct the page by viewing the network traffic, he can get access to information about application and use it to stage attacks. 
+
+**Mitigation**:
+
+1. Dashboard is not intended to be public-facing. If a decision is taken to make it public, it should be configured to use HTTPS and The [Backstage  permissions system](https://backstage.io/docs/permissions/overview) should be enabled and configured to restrict access as necessary.
+   
+**Status**:
+
+Active. Operators are expected to limit access to Dashboard portal. Further, 
+they should configure Dashboard to use HTTPS if it is public-facing.
+
+## Open Questions
+
+
+## Action Items
+
+1. Dashboard should be accessed only on HTTPS if it should be available outside cluster. Currently, we can access the application on http but since we only access the application on localhost using =kubernetes port-forward, this is OK. 
+ 
+2. Enable authentication on Dashboard. This could be tied to RBAC support on Radius, since we might want the same users to be allowed dashboard logins by default with permissions configured using Backstage permission system.
```

Comment:
What's the actual action item? Is this something the user needs to do? If that's the case what's the action item for us?

Reply:
Add documentation in dashboard related info to capture this.

Reply:
create a story